File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    var answer = 0;
    const visited = Array.from({length : n}, () => false)
    // visited[i] 가 false면 bfs 시작
    // 1일 때 2, 3 방문하는지 비교
    // 갔는데 2->1 가르키면 true
    
    function bfs(num){
        visited[num] = true
        
        for(let i = 0; i < n; i++){
            if(!visited[i] && computers[num][i] === 1){
                bfs(i)
            }
        }
    }
    
    for(let i = 0; i < n; i++){
        if(!visited[i]){
            bfs(i)
            answer++;
        }
    }
    
    return answer;
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
def dfs(start, computers, visited) :
    visited[start] = True
    
    for i in range(len(computers)) :
        if computers[start][i] == 1 and visited[i] == False :
            dfs(i, computers, visited)
    

def solution(n, computers):
    answer = 0
    visited = [False] * n
    # 우선 i,j번째 배열에 1이 있다면 네트워크 1개
    # 추가 조건으로 i j 가 있고 j k 가 연결되어도 네트워크 1개로 취급
    for i in range(n) :
        if(visited[i]) : continue
        dfs(i, computers, visited)
        answer += 1
    
    return answer